File: src/main/java/MusicCube/entities/Location.java
package MusicCube.entities;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

@Entity
public class Location {

    @Id
    @GeneratedValue
    @Column
    private int id;

    @Column
    private String country;

    @Column
    private String city;

    public Location() {}

    public Location(String country, String city) {
        this.country = country;
        this.city = city;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }
}

File: src/app/Class/Location.ts
export class Location {

    private id: number;
    private country: string;
    private city: string;

    constructor(obj?: any) {
        this.country = (obj && obj.country) || '';
        this.city = (obj && obj.city) || '';
    }

    // Getters

    getId(): number {
        return this.id;
    }

    getCountry(): string {
        return this.country;
    }

    getCity(): string {
        return this.city;
    }

    // Setters

    setCountry(country: string) {
        this.country = country;
    }

    setCity(city: string) {
        this.city = city;
    }

}
